### PR TITLE
[Codegen] Lower `hal.interface.workgroup.size` in GPU codegen

### PR DESCRIPTION
**Pull request by nithinsubbiah**
Cleanup related to https://github.com/iree-org/iree/issues/16554

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp [hunk 1]
```diff
@@ -535,6 +535,8 @@ void populateConvertSharedMemoryAllocOps(RewritePatternSet &patterns) {
 void populateLowerHALInterfaceOp(RewritePatternSet &patterns) {
   patterns.insert<HALInterfaceWorkgroupOpsConverter<
```
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp]:
```suggestion
  patterns.add<HALInterfaceWorkgroupOpsConverter<
```

**Review by kuhar — APPROVED** [review]

**Review by qedawkins — APPROVED** [review]
worth adding a test?

**Comment by antiagainst** [comment]
+1. Let's have a test before landing.

**Comment by qedawkins** [comment]
A short test here: https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMGPU/test/convert_to_rocdl.mlir#L131 (or convert to NVVM) and here: https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/SPIRV/test/convert_to_spirv.mlir

Should be good enough. Just making sure they get lowered to llvm/spirv properly.

**Review by qedawkins — APPROVED** [review]

**Pull request merged**